src/blockchain/log-processors/order-filled.ts

```diff
@@ -1,17 +1,167 @@
+      if (err) return callback(err);
+      if (!tokensRows || !tokensRows.length) return callback(new Error("market and outcome not found"));
+      const { marketID, outcome } = tokensRows[0];
+      trx.select(["minPrice", "maxPrice", "numTicks"]).from("markets").where({ marketID }).asCallback((err: Error|null, marketsRows?: Array<MarketsRow>): void => {
+        if (err) return callback(err);
+        if (!marketsRows || !marketsRows.length) return callback(new Error("market min price, max price, and/or num ticks not found"));
+        const { minPrice, maxPrice, numTicks } = marketsRows[0];
+        const price = denormalizePrice(minPrice, maxPrice, convertFixedPointToDecimal(log.price, numTicks));
+        const numCreatorTokens = convertFixedPointToDecimal(log.numCreatorTokens, WEI_PER_ETHER);
+        const numCreatorShares = convertFixedPointToDecimal(log.numCreatorShares, numTicks);
+        const numFillerTokens = convertFixedPointToDecimal(log.numFillerTokens, WEI_PER_ETHER);
+        const numFillerShares = convertFixedPointToDecimal(log.numFillerShares, numTicks);
+        const settlementFees = convertFixedPointToDecimal(log.settlementFees, WEI_PER_ETHER);
+        const orderID = log.orderId;
+        const dataToInsert: {} = {
+          marketID,
+          outcome,
+          orderID,
+          creator: log.creator,
+          filler: log.filler,
+          tradeBlockNumber: log.blockNumber,
+          tradeGroupID: log.tradeGroupID,
+          numCreatorTokens,
+          numCreatorShares,
+          numFillerTokens,
+          numFillerShares,
+          price,
+          shares: calculateNumberOfSharesTraded(numCreatorShares, numCreatorTokens, price),
+          settlementFees,
+          tradeTime: timestamp,
+        };
+        db.transacting(trx).insert(dataToInsert).into("trades").asCallback((err: Error|null): void => {
+          if (err) return callback(err);
+          const shareTokenPayload: {} = { tx: { to: log.shareToken } };
+          parallel({
+            amount: (next: AsyncCallback): void => augur.api.Orders.getAmount(orderID, next),
+            creatorPositionInMarket: (next: AsyncCallback): void => augur.trading.getPositionInMarket({ market: marketID, address: log.creator }, next),
+            fillerPositionInMarket: (next: AsyncCallback): void => augur.trading.getPositionInMarket({ market: marketID, address: log.filler }, next),
+          }, (err, onContractData): void => {
+            if (err) return callback(err);
+            const { amount, creatorPositionInMarket, fillerPositionInMarket } = onContractData!;
+            const amountRemainingInOrder = new BigNumber(amount!, 16).toFixed();
+            let updateOrdersQuery = db("orders").transacting(trx).where({ orderID });
+            if (amountRemainingInOrder === "0") {
+              updateOrdersQuery = updateOrdersQuery.del();
```

Comment:
We'll need to think about what to do here when we implement blockstream reorgs

Reply:
Yeah :grimacing: -- I think the log processors are basically just gonna need equal-and-opposite "log de-processors" that reverse the changes -- I can't think of a way to do it that's more general / doesn't require special-purpose logic for each log processor.

Reply:
I think that actually deleting a row makes it more complicated (impossible maybe?). If we kept state updates in separate tables, and linked via primary key, we could just  re-link the the prior version (and delete the now-orphaned row in the linked table). That's what I did with `market_state` table. I could imagine some helper functions that acted on the two tables fairly generically, but yea, i can't picture a more automated way of accomplishing that.